sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -54,6 +54,16 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
   val limit: Int
   override def output: Seq[Attribute] = child.output
 
+  // Do not enable whole stage codegen for a single limit.
+  override def supportCodegen: Boolean = child match {
+    case plan: CodegenSupport => plan.supportCodegen
+    case _ => false
+  }
+
+  override def executeTake(n: Int): Array[InternalRow] = child.executeTake(math.min(n, limit))
+
+  override def executeCollect(): Array[InternalRow] = child.executeTake(limit)
```

Comment:
Seems this fix relies `CollectLimitExec.executeCollect` to call `LocalLimitExec.executeTake`. Looks like we don't need to change `executeCollect`?